spec/govuk_error/configuration_spec.rb

```diff
@@ -12,36 +12,68 @@
   end
 
   describe ".should_capture" do
-    around do |example|
-      ClimateControl.modify GOVUK_DATA_SYNC_PERIOD: "22:00-08:00" do
-        example.run
-      end
-    end
-
     let(:configuration) { GovukError::Configuration.new(Raven.configuration) }
 
-    it "should capture errors that occur during the data sync" do
-      travel_to(Time.current.change(hour: 23)) do
+    context "during the data sync" do
+      around do |example|
+        ClimateControl.modify GOVUK_DATA_SYNC_PERIOD: "22:00-08:00" do
+          travel_to(Time.current.change(hour: 23)) do
+            example.run
+          end
+        end
+      end
+
+      it "should capture errors by default" do
         expect(configuration.should_capture.call(StandardError.new)).to eq(true)
       end
-    end
 
-    it "should ignore errors that have been added to data_sync_excluded_exceptions, if they occurred during the data sync" do
-      configuration.data_sync_excluded_exceptions << "StandardError"
+      it "should ignore errors that have been added to data_sync_excluded_exceptions" do
+        configuration.data_sync_excluded_exceptions << "StandardError"
 
-      travel_to(Time.current.change(hour: 23)) do
         expect(configuration.should_capture.call(StandardError.new)).to eq(false)
       end
+
+      it "should ignore errors whose underlying cause is an exception in data_sync_excluded_exceptions" do
+        stub_const("PG::Error", Class.new(StandardError))
```

Comment:
This `PG::Error` stands out to me as odd as all the other classes we're using for exceptions are made up ones. I'd suggest it being another made up one for consistency with the rest of the test.